sdk/python/kfp/compiler/_default_transformers.py

```diff
@@ -17,7 +17,7 @@
 def add_pod_env(op: BaseOp) -> BaseOp:
     """Adds pod environment info to ContainerOp.
     """
-    if isinstance(op, ContainerOp) and op.pod_labels and op.pod_labels['add-pod-env'] == 'true':
+    if isinstance(op, ContainerOp) and op.pod_labels and 'add-pod-env' in op.pod_labels and op.pod_labels['add-pod-env'] == 'true':
```

Comment:
It can be just `op.get('pod_labels', None) and op.pod_labels.get('add-pod-env'], None == 'true')`